packages/vite/src/request.ts

```diff
@@ -28,7 +28,20 @@ export class RollupModuleRequest implements ModuleRequest {
 
       // strip query params off the importer
       let fromFile = cleanUrl(nonVirtual);
-      return new RollupModuleRequest(context, source, fromFile, custom?.embroider?.meta, false, undefined);
+
+      // strip query params off the source but keep track of them
+      let cleanSource = cleanUrlQueryParams(source);
+      let queryParams = getUrlQueryParams(source);
```

Comment:
@ef4 I tried to replace these regexp-based methods with a URL object, but it caused trouble because the `source` can be a relative path. Here are examples of specifiers I get with both methods:
```
URL   : /app, 
Regexp: ../app, 

URL   : /-embroider-implicit-modules.js, 
Regexp: ./-embroider-implicit-modules.js, 

URL   : /packages/macros/src/addon/es-compat2, 
Regexp: ../../../../../../packages/macros/src/addon/es-compat2, 
```